src/dll/ScriptCompiler.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include <scc.h>
+
+class ScriptCompilerSourceRef
+{
+public:
+    uint8_t GetType() const;
+    std::string_view GetName() const;
+    std::string_view GetParentName() const;
+    std::string_view GetPath() const;
+    size_t GetLine() const;
+    bool IsNative() const;
+
+private:
+    ScriptCompilerSourceRef(SccApi& api, SccOutput* output, SccSourceRef* sourceRef);
+
+    SccApi& m_scc;
+    SccOutput* m_output;
+    SccSourceRef* m_sourceRef;
+
+    friend class ScriptCompilerOutput;
+};
+
+class ScriptCompilerOutput
+{
+    // Key used to prevent construction of this class outside of ScriptCompilerSettings.
```

Comment:
Doesn't it work if you make the ctor private?

Reply:
it's meant to be private, but I can't actually make it private because otherwise it's not possible to instantiate a std::variant in-place

Reply:
it'll give you a compilation error

Reply:
it's something to do with forwarding the arguments to the constructor, it actually gets called somewhere else, not in the friend class and it says there's no constructor

Reply:
Stupid C++ :(